src/proxy.rs

```diff
@@ -407,6 +410,38 @@ pub async fn relay(
     }
 }
 
+// pick_upstream_service selects an upstream service for inbound metrics.
+// There may be many services for a single workload. We find the the first one with an applicable port
+// as a best guess.
+pub fn pick_upstream_service(
```

Comment:
One last suggestion is better to rename this function, this is only for inbound usage. We should prevent outbound usage by mistake

Reply:
done